z3/src/z3_factory.cpp

```diff
@@ -1,15 +1,48 @@
+/*********************                                                        */
+/*! \file z3_factory.cpp
+** \verbatim
+** Top contributors (to current version):
+**   Amalee Wilson
+** This file is part of the smt-switch project.
+** Copyright (c) 2020 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Factory for creating a Z3 SmtSolver
+**
+**
+**/
+
 #include "z3_factory.h"
 
 #include "z3_solver.h"
 
-namespace smt
-{
+#include "logging_solver.h"
+
+namespace smt {
 
-/* Z3SolverFactory implementation */
-SmtSolver Z3SolverFactory::create()
+bool initialized = false;
```

Comment:
```suggestion
```

I don't think you need this for z3 -- if I recall correctly it was some weird situation with `yices2` that @amaleewilson had to work around

Reply:
Yes, this was a Yices specific thing.